sklearn/linear_model/tests/test_logistic.py

```diff
@@ -1984,3 +1985,33 @@ def test_warning_on_penalty_string_none():
     )
     with pytest.warns(FutureWarning, match=warning_message):
         lr.fit(iris.data, target)
+
+
+def test_lr_cv_scorer_does_not_receive_sample_weight_when_score_request_is_not_set():
+    X, y = make_classification(n_samples=10)
+    sample_weight = np.ones(len(y))
+    sample_weight[len(y) // 2 :] = 2
+    other_params = {}
+
+    scorer = get_scorer("accuracy")
+    lr_cv = LogisticRegressionCV(scoring=scorer)
+    routed_params = process_routing(
+        obj=lr_cv, method="fit", sample_weight=sample_weight, other_params=other_params
+    )
+    assert not routed_params.scorer.score
+
+
+def test_lr_cv_scorer_receives_sample_weight_when_score_request_is_set():
```

Comment:
We don't actually test that the weights are received or have an effect. Perhaps we should check that the results are different from when `sample_weight` is not requested.

Reply:
Thank you for the suggestion. I just wanted to mention that there seems to be an issue if I try to call fit on LogisticRegressionCV when requesting sample weight for the underlying scorer. The problem occurs at this point
https://github.com/scikit-learn/scikit-learn/pull/24498/files#diff-66422c7ed307653c80e5d68ed353d44d8fbb02319c428e140daff082eaf9d6cbR742. Over here given the sample weight has a length of 10 whereas y_test has a length of 2 and this causes an inconsistent number of samples error.

Reply:
Please make sure to index a subset of weights for the train/test set like we do for X and y... Not that I've checked the code now

Reply:
@jnothman I updated the sample weight to only index the test set at the required place and also updated the test. Could you kindly have a look?